resources/analysisTools/qcPipeline/bashLib.sh

```diff
@@ -124,6 +124,216 @@ stringJoin () {
+
+waitForRegisteredPids_BashSucksVersion() {
+    jobs
+    declare -a realPids=$(listPids)
+    if [[ -v realPids && ${#realPids[@]} -gt 0 ]]; then
+        wait ${realPids[@]}
+        declare EXIT_CODE=$?
+        if [[ ${EXIT_CODE} -ne 0 ]]; then
+            throw ${EXIT_CODE} "One of the following processes ended with exit code ${EXIT_CODE}: ${realPids[@]}"
+        fi
+    fi
+    pids=("$ARRAY_ELEMENT_DUMMY")
+}
+setUp_BashSucksVersion() {
+    declare -g -a -x tmpFiles=("$ARRAY_ELEMENT_DUMMY")
+    declare -g -a -x pids=("$ARRAY_ELEMENT_DUMMY")
+    initPipeEnds
+
+    # Remove all registered temporary files upon exit
+    trap cleanUp_BashSucksVersion EXIT
+}
+cleanUp_BashSucksVersion() {
+    if [[ $(isDebugSet) == "false" && -v tmpFiles && ${#tmpFiles[@]} -gt 1 ]]; then
+        for f in ${tmpFiles[@]}; do
+            if [[ "$f" == "$ARRAY_ELEMENT_DUMMY" ]]; then
+                continue
+            elif [[ -d "$f" ]]; then
+                rmdir "$f"
+            elif [[ -e "$f" ]]; then
+                rm -f "$f"
+            fi
+        done
+        tmpFiles=("$ARRAY_ELEMENT_DUMMY")
+    fi
+}
+
+# These versions only works with Bash >4.4. Prior version do not really declare the array variables with empty values and set -u results in error message.
+waitForRegisteredPids() {
+    jobs
+    wait ${pids[@]}
+    pids=()
+}
+setUp() {
+    declare -g -a -x tmpFiles=()
+    declare -g -a -x pids=()
+    initPipeEnds
+}
+cleanUp() {
+    if [[ $(isDebugSet) == "false" && -v tmpFiles && ${#tmpFiles[@]} -gt 0 ]]; then
+        for f in "${tmpFiles[@]}"; do
+            if [[ -d "$f" ]]; then
+                rmdir "$f"
+            elif [[ -e "$f" ]]; then
+                rm "$f"
+            fi
+        done
+        tmpFiles=()
+    fi
+}
+
+
+
+#####################################################################
+## Linear pipe management
+#####################################################################
+# The pipe-extension API follows this pattern:
+#
+# * pipeExtenderFunction pipeBaseName pipeStepExtension -- command
+#
+# * The pipeBaseName is the name under which the pipe (or derived from this: pipe pair) is registered.
+# * The pipeStepExtension is used to tag the actual pipe used in the specific step. Use a name describing the content flowing through the pipe,
+#   i.e. the output of the extension step you are declaring.
+# * Current pipe extender functions are extendPipe (for single pipes) and extendPipePair (for r1/r2 pipes)
+# * To work with paired pipes (r1/r2) mkPairedPipeName is used to calculate the r1_ or r2_ pipeBaseName
+
+_pipePath="$RODDY_SCRATCH"
+
+# Maintain a mapping of pipe-basenames to current pipe-ends.
+initPipeEnds() {
+    unset _pipeEnds
+    declare -Ag _pipeEnds=()
+}
+
+listPipeEnds() {
+    for i in "${!_pipeEnds[@]}"; do
+        echo "'$i' => '${_pipeEnds[$i]}'"
+    done
+}
+
+# Make a path without registering it.
+mkPipePath() {
+    local pipeName="${1:?Named-pipe base name}"
```

Comment:
Maybe a better error message? "Named pipe cannot be created: Call mkPipePath("somePipesName")" or so?